src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1689,7 +1705,10 @@ public synchronized ResultSet getTables(
         StringBuilder sql = new StringBuilder();
         sql.append("SELECT").append("\n");
         sql.append("  NULL AS TABLE_CAT,").append("\n");
-        sql.append("  NULL AS TABLE_SCHEM,").append("\n");
+        sql.append("  ")
+                .append(quote(s == null ? "main" : s))
```

Comment:
similar as above, we should return the actual schema of the retrieved PK's table, not the parameter that was passed to the function